File: service/src/main/java/com/ricardococati/service/impl/CalculaServiceImpl.java
package com.ricardococati.service.impl;

import com.ricardococati.model.dto.ControleExecucao;
import com.ricardococati.model.dto.MacdDiario;
import com.ricardococati.model.dto.MacdSemanal;
import com.ricardococati.repository.dao.CandlestickDiarioBuscarDAO;
import com.ricardococati.repository.dao.CandlestickSemanalBuscarDAO;
import com.ricardococati.repository.dao.ControleExecucaoDAO;
import com.ricardococati.repository.dao.MacdDiarioBuscarDAO;
import com.ricardococati.repository.dao.MacdSemanalBuscarDAO;
import com.ricardococati.repository.dao.MediaMovelExponencialDiarioDAO;
import com.ricardococati.service.CalculaService;
import java.util.List;
import lombok.Data;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;

@Slf4j
@Data
@Service
@RequiredArgsConstructor
public class CalculaServiceImpl implements CalculaService {

  private final CandlestickDiarioBuscarDAO diarioDAO;
  private final CandlestickSemanalBuscarDAO semanalDAO;
  private final MacdDiarioBuscarDAO macdDiarioBuscarDAO;
  private final MacdSemanalBuscarDAO macdSemanalBuscarDAO;
  private final MediaMovelExponencialDiarioDAO mediaExponencialDAO;
  private final ControleExecucaoDAO execucaoDAO;

  @Override
  public List<MacdDiario> listMacdDiarioByCodNeg(String codneg) {
    return macdDiarioBuscarDAO.listMacdByCodNeg(codneg);
  }

  @Override
  public List<MacdSemanal> listMacdSemanalByCodNeg(String codneg) {
    return macdSemanalBuscarDAO.listMacdByCodNeg(codneg);
  }

  @Override
  public ControleExecucao carregaControleExecucao() {
    return execucaoDAO.loadControleExecucao();
  }

  @Override
  public Boolean updateControleExecucaoDiario(ControleExecucao controleExecucao) {
    return execucaoDAO.updateControleExecucaoDiario(controleExecucao);
  }

  @Override
  public Boolean updateControleExecucaoSemanal(ControleExecucao controleExecucao) {
    return execucaoDAO.updateControleExecucaoSemanal(controleExecucao);
  }

}

File: service/src/main/java/com/ricardococati/service/impl/CalculaServiceImpl.kt
package com.ricardococati.service.impl

import com.ricardococati.model.entities.ControleExecucao
import com.ricardococati.model.entities.MacdDiario
import com.ricardococati.model.entities.MacdSemanal
import com.ricardococati.repository.dao.CandlestickDiarioBuscarDAO
import com.ricardococati.repository.dao.CandlestickSemanalBuscarDAO
import com.ricardococati.repository.dao.ControleExecucaoDAO
import com.ricardococati.repository.dao.MacdDiarioBuscarDAO
import com.ricardococati.repository.dao.MacdSemanalBuscarDAO
import com.ricardococati.repository.dao.MediaMovelExponencialDiarioBuscarDAO
import com.ricardococati.service.CalculaService
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Service

@Service
class CalculaServiceImpl(val diarioDAO: CandlestickDiarioBuscarDAO,
                         val semanalDAO: CandlestickSemanalBuscarDAO,
                         val macdDiarioBuscarDAO: MacdDiarioBuscarDAO,
                         val macdSemanalBuscarDAO: MacdSemanalBuscarDAO,
                         val mediaExponencialDAO: MediaMovelExponencialDiarioBuscarDAO,
                         val execucaoDAO: ControleExecucaoDAO
) : CalculaService {

    override fun listMacdDiarioByCodNeg(codneg: String): List<MacdDiario> {
        return macdDiarioBuscarDAO.listMacdByCodNeg(codneg)
    }

    override fun listMacdSemanalByCodNeg(codneg: String): List<MacdSemanal> {
        return macdSemanalBuscarDAO.listMacdByCodNeg(codneg)
    }

    override fun carregaControleExecucao(): ControleExecucao {
        return execucaoDAO.loadControleExecucao()
    }

    override fun updateControleExecucaoDiario(controleExecucao: ControleExecucao): Boolean {
        return execucaoDAO.updateControleExecucaoDiario(controleExecucao)
    }

    override fun updateControleExecucaoSemanal(controleExecucao: ControleExecucao): Boolean {
        return execucaoDAO.updateControleExecucaoSemanal(controleExecucao)
    }

    override fun toString(): String {
        return ("CalculaServiceImpl(diarioDAO=" + diarioDAO + ", semanalDAO=" + semanalDAO + ", macdDiarioBuscarDAO=" + macdDiarioBuscarDAO
                + ", macdSemanalBuscarDAO=" + macdSemanalBuscarDAO + ", mediaExponencialDAO="
                + mediaExponencialDAO + ", execucaoDAO=" + execucaoDAO + ")")
    }

    companion object {
        private val log = LoggerFactory.getLogger(CalculaServiceImpl::class.java)
    }

}